### PR TITLE
perf: reuse redirect handler

### PR DESCRIPTION
**Pull request by ronag**

### CODE REVIEW

**Review by dnlup — APPROVED** [review]
I am ok with the changes. I don't understand where the handler is reused.

File: lib/handler/redirect.js [hunk 1]
```diff
@@ -92,7 +92,10 @@ class RedirectHandler {
         See comment on onData method above for more detailed informations.
       */
 
-      this.agent.dispatch(this.opts, this.handler)
+      this.location = null
+      this.abort = null
+
+      this.agent.dispatch(this.opts, this)
```
Comment (ronag) [inline lib/handler/redirect.js:98]: @dnlup here it's reused. Currently it would create a new RedirectHandler in Agent.dispatch.

Comment (dnlup) [reply lib/handler/redirect.js:98]: Ahh, I missed it. Thanks.

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**